contracts/utils/NoncesKeyed.sol

```diff
@@ -7,6 +7,10 @@ import {Nonces} from "./Nonces.sol";
  * @dev Alternative to {Nonces}, that supports key-ed nonces.
  *
  * Follows the https://eips.ethereum.org/EIPS/eip-4337#semi-abstracted-nonce-support[ERC-4337's semi-abstracted nonce system].
+ *
+ * Note: This contract extends {Nonces}, and reuses {Nonces}'s storage for key #0. This makes upgrading from {Nonces}
+ * to {NoncesKeyed} safe, as such an upgrade will NOT reset the current state of nonces. Doing otherwise would enable
+ * replay attacks where operation that used nonces before the upgrade would be replayable after the upgrade.
```

Comment:
This should be clearer

```suggestion
 * NOTE: This contract extends from {Nonces} and reuses its storage for the first nonce key (i.e. `0`). This
 * makes upgrading from {Nonces} to {NoncesKeyed} safe. Doing so will NOT reset the current state of nonces to
 * avoid replay attacks where a nonce is reused after the upgrade if it resets the nonces for the first key.
```

Reply:
@arr00 is "to extend from something" correct to you? I'd say "to extend something" or "to inherit from something".

Reply:
We can just leave it like this:

```suggestion
 * NOTE: This contract extends from {Nonces} and reuses its storage for the first nonce key (i.e. `0`). This
 * makes upgrading from {Nonces} to {NoncesKeyed} safe. Doing so will NOT reset the current state of nonces to
 * avoid replay attacks where a nonce is reused after the upgrade.
```

Reply:
> @arr00 is "to extend from something" correct to you? I'd say "to extend something" or "to inherit from something".

Agree with Hadrien on these sounding more natural.